File: backend/app/models/material.py
from sqlalchemy import Column, Integer, String, Float
from sqlalchemy.orm import relationship
from app.db.database import Base
from app.models.inventory_log import InventoryChangeLog  # New import

class Material(Base):
    __tablename__ = "materials"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, unique=True, index=True, nullable=False)
    description = Column(String, nullable=True)
    quantity = Column(Float, default=0.0)  # In yards, meters, or units
    unit = Column(String, default="yards")  # e.g., yards, meters, pieces
    reorder_point = Column(Float, default=0.0)  # minimum required quantity

    # 🔁 Inventory change logs
    inventory_logs = relationship(
        "InventoryChangeLog",
        back_populates="material",
        cascade="all, delete"
    )

File: frontend/lib/models/material.dart
class Material {
  final int id;
  final String name;
  final double quantity;
  final String unit;
  final double reorderPoint;

  Material({
    required this.id,
    required this.name,
    required this.quantity,
    required this.unit,
    required this.reorderPoint,
  });

  factory Material.fromJson(Map<String, dynamic> json) {
    return Material(
      id: json['id'],
      name: json['name'],
      quantity: json['quantity'],
      unit: json['unit'],
      reorderPoint: json['reorder_point'],
    );
  }
}
